src/jit/morph.cpp

```diff
@@ -2045,12 +2045,9 @@ GenTreePtr Compiler::fgMakeTmpArgNode(
             {
                 // ToDo-ARM64: Consider using:  arg->ChangeOper(GT_LCL_FLD);
                 // as that is how FEATURE_UNIX_AMD64_STRUCT_PASSING works.
-                // We will create a GT_OBJ for the argument below.
-                // This will be passed by value in two registers.
-                assert(addrNode != nullptr);
 
-                // Create an Obj of the temp to use it as a call argument.
-                arg = gtNewObjNode(lvaGetStruct(tmpVarNum), arg);
+                // Pass by value in two registers, using a regular GT_OBJ node, created below.
+                arg->gtFlags |= GTF_DONT_CSE;
```

Comment:
I'm not sure why it would be necessary to set GTF_DONT_CSE on a GT_LCL_VAR node as such nodes are never considered CSE candidates.
If you understand why this method is setting this flag can you add a comment to the method header.